lib/web_ui/lib/src/engine/surface/backdrop_filter.dart

```diff
@@ -97,6 +126,28 @@ class PersistedBackdropFilter extends PersistedContainerSurface
     super.update(oldSurface);
     if (filter != oldSurface.filter) {
       apply();
+    } else {
+      _checkForUpdatedAncestorClipElement();
+    }
+  }
+
+  void _checkForUpdatedAncestorClipElement() {
+    // If parent clip element has moved, adjust bounds.
+    PersistedContainerSurface parentSurface = parent;
+    html.HtmlElement parentElement = null;
+    while (parentSurface != null) {
```

Comment:
This looks like it will be an infinite loop unless the immediate parent is a clip. I think there should be `parentSurface = parentSurface.parent` somewhere inside the loop. Would also be good to add a test that checks that we can climb up the parent chain.

Reply:
Done.